### PR TITLE
Tests/Inactive Cloud Watch Log Group

### PR DESCRIPTION
**Pull request by nicolasrbr**

### CODE REVIEW

**Review by Geraldo155 — APPROVED** [review]
The code is pretty neat and covers all the scenarios that I could think are possible on a unit level.

File: bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py [hunk 1]
```diff
@@ -0,0 +1,554 @@
...
+         "timestamp" : "2025-11-04T22:01:00+00:00",
+          "value" : 500
+        }]
+        resource["meta"]["metrics"]["query"] = [{
+        "timestamp" : "2025-10-31T17:55:00+00:00",
+        "value" : 1500
+        }]
+        assert mod._is_inactive(resource, 7) is False
+
+    def test_querry_and_ingestion_outside_threshold(self, module_factory):
```
Comment (caetanobca) [inline bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py]:
```suggestion
    def test_query_and_ingestion_outside_threshold(self, module_factory):
```

File: bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+        return mod
+
+    return _factory
+
+@pytest.fixture
+def mod_base(module_factory):
+    """ Basic module with default options."""
+    mod = module_factory()
+    mod.get_options_values = Mock(return_value=(7, set(), set()))
+    return mod
```
Comment (caetanobca) [inline bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py:88]: The comments in this method, as well as in others, are not very descriptive. It would be better to clearly state the behavior and configuration being applied.

For example:

```
"""
Basic module instance with default configuration.
Returns a module with a 7-day threshold and no exclusions.
"""

```

File: bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+
+class TestHasRecentMetrics:
+    """Tests for `_has_recent_metrics` method."""
+
+    def test_no_metrics(self, module_factory):
+        """No metrics."""
+        mod = module_factory()
+        series = []
+        assert mod._has_recent_metrics(series, days_threshold=7) is False
+        assert mod._count_occurrences_in_threshold(series, days_threshold=7) == 0
```
Comment (caetanobca) [inline bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py]: You should define constants for these values, for example:

```
THRESHOLD_WEEK = 7
THRESHOLD_MONTH = 30

```

File: bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+        mod._is_inactive = Mock(return_value=False)
+
+        result = mod._get()
+
+        mod._aggregate_resources.assert_called_once_with("acc1")
+        assert result == []
+
+
+# Enum used in many scenarios — imported here to avoid unused symbol lint.
+_ = MetricKey
```
Comment (caetanobca) [inline bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py]: Is this really necessary?

File: bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py
Comment (caetanobca) [inline bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py:1]: You could take a look at `@pytest.mark.parametrize`.  This could reduce code repetition in your tests.

Comment (caetanobca) [inline bumiworker/bumiworker/tests/test_inactive_cloud_watch_log_group.py:1]: I've reviewed the tests and they look good for the primary use cases. To make them even more robust, I recommend adding tests for the following edge cases:
1. Invalid or negative metric values;
2. Invalid timestamp formats.

**Review by caetanobca — CHANGES_REQUESTED** [review]

**Review by marianezei — COMMENTED** [review]
I agree with Caetano's suggestions.

LGTM!!

**Review by caetanobca — APPROVED** [review]
LGTM!!
